core/runtime/src/main/java/io/quarkus/runtime/LiveReloadConfig.java

```diff
@@ -1,20 +1,21 @@
 package io.quarkus.runtime;
 
+import java.util.List;
 import java.util.Optional;
 
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
 
-/**
- * This is used currently only to suppress warnings about unknown properties
- * when the user supplies something like: -Dquarkus.live-reload.password=secret
- *
- * TODO refactor code to actually use these values
- */
 @ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
 public class LiveReloadConfig {
 
+    /**
+     * The names of additional resources to watch for changes, triggering a reload on change.
+     */
+    @ConfigItem
+    public Optional<List<String>> additionalWatchedResources;
```

Comment:
More of a curiosity than a review comment - For types like `List` does it add value to wrap it add around a `Optional`? If no config values are provided for the property, then the config management framework can perhaps just create an empty list?

Reply:
Ah ah, you want to talk to @dmlloyd about that :).

If you want to allow the empty list, you need an `Optional`.

Reply:
I don't remember the exact reason (although David certainly does and perhaps @ia3andy to whom David had explained it recently), but what Guillaume says is spot on.

Reply:
@dmlloyd will confirm but from what I remember, it was to be consistant for all different types, it's not `Optional` if not specified as an `Optional` whatever the type. and you can always decide to convert the empty optional to an empty list when consuming it.

I have a task in my todos to document that a bit better, but it's always sliding, I need to get to it 😅, aw and I created an issue:
https://github.com/quarkusio/quarkus/issues/8306